source/manual/alerts/rabbitmq-no-consumers-listening.html.md

```diff
@@ -4,75 +4,106 @@ title: 'RabbitMQ: No consumers listening to queue'
+
+Looking at the queue graphs we should look out for the following:
+
+* Check for high 'ready' messages for the queue
+
+This indicates these messages are waiting to be processed in RabbitMQ by the
+consumer (e.g email-alert-service).
+
+* Check for high 'unacknowledged' messages for the queue
+
+This implies that messages have been read by the consumer but the consumer has
+never sent back an ACK to the RabbitMQ broker to say that it has finished
+processing.
+
+* Check for high 'redeliver' rate for the queue
 
-## Consequences of idle consumers
+In the event of network failure (or a node failure), messages can be
+redelivered. An example is if the consumer dies (its channel is closed,
+connection is closed, or TCP connection is lost).
 
-If consumers are idle, messages sent to the queue will not be being processed.
-This means that the actions which are meant to happen in response to the
-messages will not happen. For example, email alerts about updated content will
-not be sent if the `email_alert_sevice` queue isn't running.
+If we're seeing high 'redeliver' rates, high 'ready' or 'unacknowledged'
+messages then this could indicate an issue with the consumer.
 
-Unless there is a wider RabbitMQ failure, messages will not be lost - they will
-be processed once the problem is resolved.
+### Troubleshooting steps
 
-## Investigation
+1. Check the status of the consumer, e.g for email-alert-service look on the
+   email_alert_api machines - `sudo service email-alert-service status`
 
-RabbitMQ has an admin interface, which allows details of recent activity on the
-queues to be seen. To log in, [follow the instructions in the RabbitMQ
-docs][rabbitmq_control_panel].
+2. If the services are running we should look in the consumers application logs.
 
-The admin interface allows you to see details (and graphs!) of the messages
-sent to each queue, and the number of messages held on the queues. Look at the
-queue for which the alert happened. If no messages have been sent to the queue
-over the last several minutes, the most likely failure is that the heartbeat
-messages are no longer being sent correctly. Look for recent changes to the
-[Publishing API][publishing_api].
+3. If messages are being received by the consumer check they are actually being
+   actioned (e.g email-alert-api is receiving the messages email-alert-service
+   has received from RabbitMQ). To do this we could compare the content changes
+   in [email-alert-api][email-alert-api] against whats recorded in the
+   email-alert-service logs.
 
-If messages have been received, and the queue has messages held on it and not
-being processed, the consumers have either got stuck or stopped running for
-some reason. Restarting the consumers for the relevant app will probably clear
-this up (but please make sure the team owning the app are also alerted, since
-this shouldn't normally happen). For example:
+4. If messages aren't being processed then the consumers could be stuck or
```

Comment:
This could replace the first bullet point: it's worth trying a blind restart ([esp. for RabbitMQ](https://docs.publishing.service.gov.uk/manual/alerts/rebooting-machines.html#rebooting-rabbitmq-machines)) to see if that fixes it, and then seeing if there are any errors.

Reply:
Sure, I've replaced step 1.